app/controllers/application_controller.rb

```diff
@@ -10,8 +10,8 @@ class ApplicationController < ActionController::Base
 
   before_action :authenticate_http_basic, if: :http_basic_auth_site?
 
+  around_action :switch_locale
   before_action :ensure_signup_complete
```

Comment:
Would it be OK to keep it in the same order it was before these changes? :thinking:

```suggestion
  before_action :ensure_signup_complete
  around_action :switch_locale
```

Reply:
Just thinking out loud ... as we are setting the request locale I think this could be done before doing anything else, imagine that some CONSUL fork changes the code of the `ensure_signup_complete`metho to redirect the users to another page with a notice under certain conditions ... the notice translation could be shown with the wrong language 🤔 . What do you think?

Reply:
Maybe :thinking:. In that case, could we do so in a different commit? I wonder whether we could add a test for this scenario :thinking:.

By the way, earlier today I noticed the `finish_signup` action doesn't load the right locale because it uses `prepend_before_action`. So going to, for instance, https://demo.consulproject.org/finish_signup?locale=it as an anonymous user will redirect to the sign in page in English (or whatever language is stored in the `session`). Not sure how to fix that, though, and maybe we can leave with that :relieved:.